docs/core/seeding.rst

```diff
@@ -242,19 +285,34 @@ previous check, as the scattering decreases linearly with particle energy
 
 .. code-block:: cpp
 
-   // calculate scattering for p(T) calculated from seed curvature
-   float pT2scatter = 4 * iHelixDiameter2 * m_config.pT2perRadius;
-   // TODO: include upper pT limit for scatter calc
-   // convert p(T) to p scaling by sin^2(theta) AND scale by 1/sin^4(theta)
-   // from rad to deltaCotTheta
-   float p2scatter = pT2scatter * iSinTheta2;
-   // if deltaTheta larger than allowed scattering for calculated pT, skip
-   if ((deltaCotTheta2 - error2 > 0) &&
-       (dCotThetaMinusError2 >
-        p2scatter * m_config.sigmaScattering * m_config.sigmaScattering)) {
-     continue;
-   }
-
+	// refinement of the cut on the compatibility between the r-z slope of
+	// the two seed segments using a scattering term scaled by the actual
+	// measured pT (p2scatterSigma)
+	float iHelixDiameter2 = B2 / S2;
+	// calculate scattering for p(T) calculated from seed curvature
+	float pT2scatterSigma = iHelixDiameter2 * m_config.sigmapT2perRadius;
+	// if pT > maxPtScattering, calculate allowed scattering angle using
+	// maxPtScattering instead of pt.
+	float pT = m_config.pTPerHelixRadius * std::sqrt(S2 / B2) / 2.;
+	if (pT > m_config.maxPtScattering) {
+		float pTscatterSigma =
+				(m_config.highland / m_config.maxPtScattering) *
+				m_config.sigmaScattering;
+		pT2scatterSigma = pTscatterSigma * pTscatterSigma;
+	}
+	// convert p(T) to p scaling by sin^2(theta) AND scale by 1/sin^4(theta)
+	// from rad to deltaCotTheta
+	float p2scatterSigma = pT2scatterSigma * iSinTheta2;
+	// if deltaTheta larger than allowed scattering for calculated pT, skip
+	if (deltaCotTheta2 > (error2 + p2scatterSigma)) {
+		if (m_config.skipPreviousTopSP) {
+			if (cotThetaB - cotThetaT < 0) {
+				break;
+			}
+			t0 = t;
+		}
+		continue;
+	}
 
 The last cut applied in this function is on the so-called impact parameters,
```

Comment:
transverse impact parameter (or DCA - distance of closest approach) - various experiments refer to it differently